deploy/lib/xquery/setup.xqy

```diff
@@ -2538,6 +2497,127 @@ declare function setup:validate-path-namespaces(
+                        return $i/db:field-name/fn:string()
+    let $admin-config := admin:database-delete-all-range-field-indexes($admin-config, $database, $field-names)
+    return admin:database-delete-field($admin-config, $database, $field-names)
+};
+
+declare private function setup:remove-existing-path-range-indexes(
+  $admin-config as element(configuration),
+  $database as xs:unsignedLong,
+  $prefix as xs:string
+) as element(configuration)
+{
+    let $pref := $prefix||":"
+    (:find and delete those path range indexes which use the input prefix:)
+    let $path-indexes := admin:database-get-range-path-indexes($admin-config, $database)[fn:contains(./db:path-expression,$pref)]
+    return admin:database-delete-range-path-index($admin-config, $database, $path-indexes)
+};
+
+declare private function setup:remove-existing-geospatial-path-indexes(
+  $admin-config as element(configuration),
+  $database as xs:unsignedLong,
+  $prefix as xs:string
+) as element(configuration)
+{
+    let $pref := $prefix||":"
+    (:find and delete those geospatial path range indexes which use the input prefix:)
+    let $path-indexes := admin:database-get-geospatial-path-indexes($admin-config, $database)[fn:contains(./db:path-expression,$pref)]
+    return admin:database-delete-geospatial-path-index($admin-config, $database, $path-indexes)
+
+};
+
+(:~
+ :
+ : Apply Path Namespace Settings
+ :
+ : Many other artifacts in MarkLogic can be dependent on a path namespace
+ :  field definitions with paths
+ :  path range indexes
+ :  geospatial path range indexes.
+ : Therefore if a path changes or is removed we need to consider these.
+ : Up to ML 8.0-6 we unconditionally removed all path namespaces and path range indexes and added them all back in again in every bootstrap
+ : This would cause re-indexing. From 8.0-7 it was necessary to additionally remove fields depending on paths and their indexes. This would
+ : have meant considerable reindexing.
+ : This function seeks to apply the path namespace settings removing dependent artifacts only where necessary, specifically
+ : where a namespace has been changed (same prefix, different namespace uri) or removed.
+ : Paths do not exist in MarkLogic version < 6.02 and if an old version is detected the function does nothing and just
+ : returns the input configuration
+ :
+ : @param admin-config - the existing application configuration
+ : @param database - the id of the database in which the changes are to be applied
+ : @param db-config - the input configuration to be applied to this database
+ : @return the modified configuration
+ :
+:)
+declare function setup:apply-path-namespaces-settings(
+  $admin-config as element(configuration),
+  $database as xs:unsignedLong,
+  $db-config as element(db:database)
+) as element(configuration)
+{
+  (:paths were not supported before this version of ML:)
+  if (setup:at-least-version("6.0-2")) then
```

Comment:
This check is redundant. ML7- is obsolete, and below references functions that wouldn't exist in ML6, so setup.xqy wouldn't compile at all..

Reply:
Then again, minor detail..